libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/GranularPermissions/GranularPermissionsDropdownBody.tsx

```diff
@@ -63,8 +81,65 @@ export const GranularPermissionsDropdownBody: React.FC<IGranularPermissionsDropd
     onDelete,
     handleSetSelectedPermission,
     mode,
+    toggleButtonRef,
+    id,
 }) => {
     const { permissionsChangeInteraction, permissionsRemoveInteraction } = useShareDialogInteraction();
+    const dropdownRef = useRef<HTMLDivElement>(null);
+    const selectedItemRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        if (isShowDropdown && selectedItemRef.current) {
+            setTimeout(() => {
+                selectedItemRef.current?.focus();
+            }, 100);
+        }
+    }, [isShowDropdown]);
+
+    const handleKeyDown = useCallback(
```

Comment:
can we separate this keyboard navigation to use it on multiple lists? Check with @jkbmat if his Listbox component can not be used here or at least create some keyboard navigation hook which can be reused on multiple places

Reply:
Updated to use generic hook `makeMenuKeyboardNavigation`